### PR TITLE
Implement color theming support

### PR DESCRIPTION
**Pull request by jordanjones243**
# Alaska Airlines Pull Request

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

**Resolves:** #121 

* Implement tier 3 tokens for color theming support
* Update dependencies

## Summary:

_Please summarize the scope of the changes you have submitted, what the intent of the work is and anything that describes the before/after state of the project._

## Type of change:

Please delete options that are not relevant.

- [x] New capability
- [x] Revision of an existing capability
- [ ] Infrastructure change (automation, etc.)
- [ ] Other (please elaborate)

## Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

## Summary by Sourcery

Implement color theming support by introducing tier 3 tokens and updating component styles to use these tokens. Enhance documentation to include theme support details and refactor SCSS files for better maintainability.

New Features:
- Introduce tier 3 tokens for color theming support, allowing components to be restyled using new SCSS variables.

Enhancements:
- Refactor SCSS files to use shared styles for flight time elements, reducing code duplication.
- Update component styles to utilize new color theming tokens for better customization.

Documentation:
- Add documentation on theme support, including code samples for restyling components using new tokens.

Chores:
- Remove outdated CSS custom property fallback documentation from README.

### CODE REVIEW

**Comment by jason-capsule42** [comment]
@sourcery-ai review

File: src/tokens.scss [hunk 1]
```diff
@@ -0,0 +1,10 @@
+@import './../node_modules/@aurodesignsystem/design-tokens/dist/tokens/SCSSVariables';
+
+:host {
+  --ds-auro-flight-footer-text-color: var(--ds-color-text-secondary-default, #{$ds-color-text-secondary-default});
+  --ds-auro-flight-header-days-changed-text-color: var(--ds-color-utility-error-default, #{$ds-color-utility-error-default});
+  --ds-auro-flight-header-text-color: var(--ds-color-text-secondary-default, #{$ds-color-text-secondary-default});
+  --ds-auro-flight-main-cancelled-station-text-color: var(--ds-color-utility-error-default, #{$ds-color-utility-error-default});
+  --ds-auro-flight-main-station-text-color: var(--ds-color-text-secondary-default, #{$ds-color-text-secondary-default});
+  --ds-auro-flight-main-time-text-color: var(--ds-color-text-primary-default, #{$ds-color-text-primary-default});
```
Comment (jason-capsule42) [inline src/tokens.scss:9]: Can we simplify this and not use "-main" in the name?

File: package.json
Comment (jason-capsule42) [inline package.json:1]: Can you update the scripting here to use the auro-library references like we did in datepicker?

**Review by jason-capsule42 — CHANGES_REQUESTED** [review]

File: README.md [hunk 2]
```diff
@@ -58,11 +58,12 @@ For the most up to date information on [UI development browser support](https://
 
 ## Install
 
-<!-- AURO-GENERATED-CONTENT:START (REMOTE:url=https://raw.githubusercontent.com/AlaskaAirlines/WC-Generator/master/componentDocs/partials/usage/componentInstall.md) -->
+<!-- AURO-GENERATED-CONTENT:START (REMOTE:url=https://raw.githubusercontent.com/AlaskaAirlines/WC-Generator/master/componentDocs/partials/usage/componentInstall_esm.md) -->
```
Comment (jason-capsule42) [inline README.md:61]: I just want to confirm that this component is actually ESM ready since we weren't using that template before.

Comment (jordanjones243) [reply README.md:61]: According to this, this component does support ESM:
- https://github.com/orgs/AlaskaAirlines/discussions/502

**Review by jason-capsule42 — APPROVED** [review]

**Pull request merged**